File: app/src/main/java/com/ashlikun/core/simple/MainActivity.java
package com.ashlikun.core.simple;

import android.content.Intent;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.util.Log;

import com.ashlikun.core.activity.BaseMvpActivity;

public class MainActivity extends BaseMvpActivity<MainPresenter>
        implements IMainView {


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState, @Nullable PersistableBundle persistentState) {
        super.onCreate(savedInstanceState, persistentState);
        Log.e("MainActivity", "onCreate");
    }

    @Override
    public int getLayoutId() {
        Log.e("MainActivity", "getLayoutId");
        return R.layout.activity_main;

    }

    @Override
    public void initView() {
        Log.e("MainActivity", "initView");
        getSupportFragmentManager().beginTransaction()
                .add(R.id.fragment, new MainFragment()).commit();
    }

    @Override
    public void parseIntent(Intent intent) {
        Log.e("MainActivity", "parseIntent" + (presenter == null));
    }

    @Override
    public void clearData() {

    }
}

File: app/src/main/java/com/ashlikun/core/simple/MainActivity.kt
package com.ashlikun.core.simple

import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log

import com.ashlikun.core.activity.BaseMvpActivity

class MainActivity : BaseMvpActivity<MainPresenter>(), IMainView {


    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        Log.e("MainActivity", "onCreate")
    }

    override fun getLayoutId(): Int {
        Log.e("MainActivity", "getLayoutId")
        return R.layout.activity_main

    }

    override fun initView() {
        Log.e("MainActivity", "initView")
        supportFragmentManager.beginTransaction()
                .add(R.id.fragment, MainFragment()).commit()
    }

    override fun parseIntent(intent: Intent) {
        Log.e("MainActivity", "parseIntent" + (presenter == null))
    }

    override fun clearData() {

    }

    override fun findSize() = 111
}
